### PR TITLE
Fix sequential stage views in parallel nodes for Blue Ocean

### PR DESCRIPTION
**Pull request by eidenyoshida**
http://10.216.151.18:8080/blue/organizations/jenkins/ROCmSoftwarePlatform%2FrocRAND/detail/PR-96/16/pipeline

### CODE REVIEW

File: src/com/amd/docker/dockerNodes.groovy [hunk 1]
```diff
@@ -26,7 +26,7 @@ class dockerNodes implements Serializable
         {
             if(it.contains('cuda'))
             {
-                dockerArray[it] = new rocDocker(
+                dockerArray[it + " && " + rocmVersion] = new rocDocker(
```
Comment (saadrahim) [inline src/com/amd/docker/dockerNodes.groovy]: why not change the iterator to have the rocmVersion

File: vars/buildProject.groovy [hunk 2]
```diff
@@ -185,164 +226,135 @@ def call(rocProject project, boolean formatCheck, def dockerArray, def compileCo
...
+                            
+                            if(platform.jenkinsLabel.contains('hip-clang'))
                             {
-                                timeout(time: project.timeout.permissions, unit: 'MINUTES')
-                                {
-                                    startTime = project.email.start()
-                                    permissionsCommand = "sudo chown jenkins -R ./*"
-                                    platform.runCommand(this, permissionsCommand)
-                                }
+                                recipient = 'akila.premachandra@amd.com'
```
Comment (saadrahim) [inline vars/buildProject.groovy]: This should be email list, personal emails not allowed

File: vars/buildProject.groovy [hunk 3]
```diff
@@ -185,164 +226,135 @@ def call(rocProject project, boolean formatCheck, def dockerArray, def compileCo
...
+                                            Job: ${platform.jenkinsLabel}
+                                            <br>Node: ${env.NODE_NAME}
+                                            <br><br>View ${project.name} ${env.BRANCH_NAME}:    ${link}
+                                            <br>View the full log:  ${log}
+                                            <br>View pipeline steps:    ${stageView}
+                                            <br>View in Blue Ocean:    ${blueOcean}
+                                        """,
+                                    cc: '',
+                                    charset: 'UTF-8',
+                                    from: 'dl.mlse.lib.jenkins@amd.com',
```
Comment (saadrahim) [inline vars/buildProject.groovy]: use this email above

File: src/com/amd/docker/dockerNodes.groovy (hunk 1 above)
Comment (amdkila) [reply src/com/amd/docker/dockerNodes.groovy]: That requires editing all the Jenkinsfiles. dockerArray is filled when calling buildProject in the Jenkinsfile.

File: vars/buildProject.groovy [hunk 4]
```diff
@@ -185,164 +226,135 @@ def call(rocProject project, boolean formatCheck, def dockerArray, def compileCo
...
+                                stageView = "${env.JENKINS_URL}job/ROCmSoftwarePlatform/job/${project.name}/job/${env.BRANCH_NAME}/${env.BUILD_NUMBER}/flowGraphTable"              
+                                blueOcean = "${env.JENKINS_URL}blue/organizations/jenkins/ROCmSoftwarePlatform%2F${project.name}/detail/${env.BRANCH_NAME}/${env.BUILD_NUMBER}/pipeline"
+                                recipient = "dl.${project.name}-ci@amd.com"
+                            }
+                    
+                            stageTime = project.email.timeFunction(duration)
+                            
+                            if(platform.jenkinsLabel.contains('hip-clang'))
+                            {
+                                recipient = 'akila.premachandra@amd.com'
```
Comment (saadrahim) [inline vars/buildProject.groovy]: email list

File: vars/runParallelStage.groovy [hunk 5]
```diff
@@ -5,27 +5,32 @@
...
 
         node (platform.jenkinsLabel)
         {
-            stage ("${platform.jenkinsLabel}") 
+            stage ("${platform.jenkinsLabel}")
             {
-                body(platform, runCode)
+                for (int i = 0; i < stages.size(); i++) {
+                    stage ("${stages[i]}") 
+			        {
```
Comment (saadrahim) [inline vars/runParallelStage.groovy]: Doesn't look correctly formatted on github, may be correct.

File: src/com/amd/docker/dockerNodes.groovy (hunk 1 above)
Comment (saadrahim) [reply src/com/amd/docker/dockerNodes.groovy]: Can we not add it afterwards? we are cutting pasting this in multiple locations?

Comment (saadrahim) [reply src/com/amd/docker/dockerNodes.groovy]: I guess we can keep it this way and refactor this section in a later PR. The if else statements are not too robust and lots of duplication.

**Pull request merged**